### PR TITLE
Review

### PR DESCRIPTION
**Pull request by bjsowa**

### CODE REVIEW

File: leo_gz_bringup/package.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>ament_cmake_copyright</test_depend>
+  <test_depend>ament_cmake_lint_cmake</test_depend>
+  <test_depend>ament_cmake_mypy</test_depend>
+  <test_depend>ament_cmake_xmllint</test_depend>
+
+  <export>
+    <build_type>ament_cmake</build_type>
```
Comment (bjsowa) [inline leo_gz_bringup/package.xml:32]: Add `architecture_independent` tag

File: leo_gz_plugins/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    gz-sim${GZ_SIM_VER}::gz-sim${GZ_SIM_VER}
+    gz-plugin${GZ_PLUGIN_VER}::gz-plugin${GZ_PLUGIN_VER}
+  )
+else()
+  #If GZ_VERSION is not set default to Fortress
+  find_package(ignition-plugin1 REQUIRED COMPONENTS register)
+  set(IGN_PLUGIN_VER ${ignition-plugin1_VERSION_MAJOR})
+
+  find_package(ignition-common4 REQUIRED)
+  set(IGN_COMMON_VER ${ignition-common4_VERSION_MAJOR})
```
Comment (bjsowa) [inline leo_gz_plugins/CMakeLists.txt]: We don't include any header from the common library so linking it explicitely is redundant
```suggestion
```

File: leo_gz_plugins/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+src/differential_system.cpp)
+
+if(DEFINED ENV{GZ_VERSION} AND "$ENV{GZ_VERSION}" STREQUAL "garden")
+  find_package(gz-plugin2 REQUIRED COMPONENTS register)
+  set(GZ_PLUGIN_VER ${gz-plugin2_VERSION_MAJOR})
+
+  find_package(gz-sim7 REQUIRED)
+  set(GZ_SIM_VER ${gz-sim7_VERSION_MAJOR})
+
+  target_link_libraries(leo_gazebo_differential_plugin
```
Comment (bjsowa) [inline leo_gz_plugins/CMakeLists.txt]: Rename to `leo_gz_differential_plugin` to follow the naming convention (gazebo -> gazebo classic, gz -> new gazebo)

File: leo_gz_plugins/package.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    <license>MIT</license>
+    <author email="janek@fictionlab.pl">Jan Hernas</author>
+
+    <buildtool_depend>ament_cmake</buildtool_depend>
+    <!-- Garden -->
+    <depend condition="$GZ_VERSION == garden or $IGNITION_VERSION == garden">gz-plugin2</depend>
+    <depend condition="$GZ_VERSION == garden or $IGNITION_VERSION == garden">gz-sim7</depend>
+    <!-- Fortress (default) -->
+    <depend condition="$GZ_VERSION == fortress or $IGNITION_VERSION == fortress">ignition-plugin1</depend>
+    <depend condition="$GZ_VERSION == fortress or $IGNITION_VERSION == fortress">ignition-common4</depend>
```
Comment (bjsowa) [inline leo_gz_plugins/package.xml]:
```suggestion
```

File: leo_gz_bringup/package.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  <version>1.1.0</version>
+
+  <description>Bringup package for Leo Rover Gazebo simulation in ROS 2</description>
+
+  <maintainer email="support@fictionlab.pl">Fictionlab</maintainer>
+  <license>MIT</license>
+  <author email="janek@fictionlab.pl">Jan Hernas</author>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
```
Comment (bjsowa) [inline leo_gz_bringup/package.xml:14]: Add missing dependencies
```suggestion

  <exec_depend>ament_index_python</exec_depend>
  <exec_depend>robot_state_publisher</exec_depend>
  <exec_depend>ros_gz_bridge</exec_depend>
  <exec_depend>ros_gz_image</exec_depend>
  <exec_depend>ros_gz_sim</exec_depend>
  <exec_depend>xacro</exec_depend>

  <exec_depend>leo_gz_plugins</exec_depend>
  <exec_depend>leo_gz_worlds</exec_depend>

```

File: leo_gz_plugins/src/differential_system.cpp [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+#else
+#define ADD_PLUGIN GZ_ADD_PLUGIN
+#include <gz/sim/System.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/components.hh>
+#include <gz/plugin/Register.hh>
+#define  gazebo gz::sim
+#define error gzerr
+#define msg gzmsg
+#endif
```
Comment (bjsowa) [inline leo_gz_plugins/src/differential_system.cpp]: Gazebo garden provides backward compatibility with the ignition headers and macros. I think we should make it simpler by using the compatibility as long as we support Gazebo Fortress.

My proposal:
```suggestion
#define SUPPRESS_IGNITION_HEADER_DEPRECATION

#include <ignition/gazebo/System.hh>
#include <ignition/gazebo/Model.hh>
#include <ignition/gazebo/components.hh>
#include <ignition/plugin/Register.hh>

#ifdef USE_IGNITION
namespace gazebo = ignition::gazebo;
#else
namespace gazebo = gz::sim;
#endif

```

File: leo_gz_worlds/package.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    <author email="janek@fictionlab.pl">Jan Hernas</author>
+
+    <buildtool_depend>ament_cmake</buildtool_depend>
+    <test_depend>ament_lint_auto</test_depend>
+    <test_depend>ament_cmake_copyright</test_depend>
+    <test_depend>ament_cmake_lint_cmake</test_depend>
+    <test_depend>ament_cmake_xmllint</test_depend>
+
+    <export>
+        <build_type>ament_cmake</build_type>
```
Comment (bjsowa) [inline leo_gz_worlds/package.xml:20]: Add `architecture_independent` tag

File: leo_simulator/package.xml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  <maintainer email="support@fictionlab.pl">Fictionlab</maintainer>
+
+  <license>MIT</license>
+
+  <author email="janek@fictionlab.pl">Jan Hernas</author>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <exec_depend>leo_gz_bringup</exec_depend>
+  <exec_depend>leo_gz_plugins</exec_depend>
```
Comment (bjsowa) [inline leo_simulator/package.xml:18]: Add leo_gz_worlds

File: leo_gz_plugins/package.xml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+    <name>leo_gz_plugins</name>
+    <version>1.1.0</version>
+
+    <description>Package with plugins for Leo Rover Gazebo simulation in ROS 2</description>
```
Comment (bjsowa) [inline leo_gz_plugins/package.xml]: Skip boilerplate `Package with`

```suggestion
    <description>Plugins for Leo Rover Gazebo simulation in ROS 2</description>
```

File: leo_gz_worlds/package.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+    <name>leo_gz_worlds</name>
+    <version>1.1.0</version>
+
+    <description>Package with Gazebo worlds for Leo Rover simulation in ROS 2</description>
```
Comment (bjsowa) [inline leo_gz_worlds/package.xml]: Skip boilerplate `Package with`

```suggestion
    <description>Gazebo worlds for Leo Rover simulation in ROS 2</description>
```

File: leo_gz_bringup/launch/spawn_robot.launch.py [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+        robot_state_publisher,
+        leo_rover,
+        topic_bridge,
+        image_bridge,
+    ]
+
+
+def generate_launch_description():
+    name_argument = DeclareLaunchArgument(
+        "robot_name",
```
Comment (bjsowa) [inline leo_gz_bringup/launch/spawn_robot.launch.py]: Change argument name to `robot_ns`

File: leo_gz_bringup/launch/leo_gz.launch.py [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+        PythonLaunchDescriptionSource(
+            os.path.join(pkg_ros_gz_sim, "launch", "gz_sim.launch.py")
+        ),
+        launch_arguments={"gz_args": LaunchConfiguration("sim_world")}.items(),
+    )
+
+    robot_sim = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            os.path.join(pkg_project_gazebo, "launch", "spawn_robot.launch.py")
+        ),
```
Comment (bjsowa) [inline leo_gz_bringup/launch/leo_gz.launch.py:63]: Add `robot_ns` launch argument to this launch file and pass it to spawn_robot launch

File: leo_gz_bringup/launch/leo_gz.launch.py [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+    # Setup to launch the simulator and Gazebo world
+    gz_sim = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            os.path.join(pkg_ros_gz_sim, "launch", "gz_sim.launch.py")
+        ),
+        launch_arguments={"gz_args": LaunchConfiguration("sim_world")}.items(),
+    )
+
+    robot_sim = IncludeLaunchDescription(
```
Comment (bjsowa) [inline leo_gz_bringup/launch/leo_gz.launch.py]:
```suggestion
    spawn_robot = IncludeLaunchDescription(
```

File: leo_gz_worlds/worlds/leo_empty.sdf [hunk 14]
```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0"?>
+<sdf version="1.8">
+  <world name="leo_empty">
+    <physics name="1ms" type="ignored">
+      <max_step_size>0.001</max_step_size>
+      <real_time_factor>1.0</real_time_factor>
+      <dart>
+        <!-- Heightmaps behave better with the bullet collision detector -->
+        <collision_detector>bullet</collision_detector>
```
Comment (bjsowa) [inline leo_gz_worlds/worlds/leo_empty.sdf:8]: Can this be passed through gz sim arguments instead of hardcoded in the world description?

**Comment by jasiex01** [comment]
All required changes implemented and CI is passing

**Pull request closed without merging**